src/main/java/org/sefglobal/scholarx/service/MenteeService.java

```diff
@@ -14,9 +21,39 @@
 
     private final static Logger log = LoggerFactory.getLogger(MenteeService.class);
     private final MenteeRepository menteeRepository;
+    private final MentorRepository mentorRepository;
+    private final ProgramRepository programRepository;
+    private final ProfileRepository profileRepository;
 
-    public MenteeService(MenteeRepository menteeRepository) {
+    public MenteeService(MenteeRepository menteeRepository, MentorRepository mentorRepository, ProgramRepository programRepository, ProfileRepository profileRepository) {
         this.menteeRepository = menteeRepository;
+        this.mentorRepository = mentorRepository;
+        this.programRepository = programRepository;
+        this.profileRepository = profileRepository;
+    }
+
+    /**
+     * Create new {@link Mentee}
+     *
+     * @param profileId which is the Profile id of the applying user's {@link Profile}
+     * @param mentorId which is the requested {@link Mentor}
+     * @return the created {@link Mentee}
+     * @throws ResourceNotFoundException is thrown if the applying {@link Mentor} doesn't exist
+     */
+    public Mentee applyAsMentee(long mentorId, long profileId) throws ResourceNotFoundException {
+        Optional<Mentor> optionalMentor = mentorRepository.findById(mentorId);
+        if (!optionalMentor.isPresent()) {
+            String msg = "Error, Unable to apply as a mentee. " +
+                         "Mentor with id: " + mentorId + " doesn't exist.";
+            log.error(msg);
+            throw new ResourceNotFoundException(msg);
+        }
+
+        Mentee mentee = new Mentee();
+        mentee.setProfile(profileRepository.getOne(profileId));
```

Comment:
`profileId` should be validated for the following scenario,

* Does the profile exist?

Reply:
Done